File: dsa/medium/3. Move Elements To End/move_elements_to_end.go
package main

func MoveElementToEnd(array []int, toMove int) []int {
	left := 0
	right := len(array) - 1

	for left < right {
		for array[left] != toMove && left < right {
			left++
		}

		for array[right] == toMove && left < right {
			right--
		}

		if left < right {
			array[left], array[right] = array[right], array[left]
			left++
			right--
		}
	}
	return array
}

File: dsa/medium/3. Move Elements To End/move_elements_to_end.cpp
#include <vector>

std::vector<int> moveElementToEnd(std::vector<int> array, int toMove) {
    int left = 0;
    int right = array.size() - 1;
    while (left < right) {
        while (array[left] != toMove && left < right) {
            left++;
        }
        while (array[right] == toMove && left < right) {
            right--;
        }
        if (left < right) {
            std::swap(array[left], array[right]);
        }
    }
    return array;
}

